src/mm/memory_manager.rs

```diff
@@ -96,7 +96,7 @@ pub fn alloc<VM: VMBinding>(
     size: usize,
     align: usize,
     offset: isize,
-    allocator: Allocator,
+    allocator: AllocationSemantic,
```

Comment:
Maybe we should also change `allocator` e.g. to `semantics` in the future?

Reply:
Yeah. We will need at least another PR for https://github.com/mmtk/mmtk-core/issues/142.